swift/ql/test/query-tests/Security/CWE-760/test.swift

```diff
@@ -0,0 +1,70 @@
+
+// --- stubs ---
+
+// These stubs roughly follows the same structure as classes from CryptoSwift
+enum PKCS5 { }
+
+enum Variant { case md5, sha1, sha2, sha3 }
+
+extension PKCS5 {
+  struct PBKDF1 {
+	init(password: Array<UInt8>, salt: Array<UInt8>, variant: Variant = .sha1, iterations: Int = 4096, keyLength: Int? = nil) { }
+  }
+
+  struct PBKDF2 {
+	init(password: Array<UInt8>, salt: Array<UInt8>, iterations: Int = 4096, keyLength: Int? = nil, variant: Variant = .sha2) { }
+  }
+}
+
+struct HKDF {
```

Comment:
Looking at https://developer.apple.com/documentation/cryptokit/hkdf it seems like the _real_ struct is a generic one. Does the query also work on the real thing? Should we make this stub more realistic?

Reply:
We could, but this class is actually not the one in CryptoKit. It's the one in [CryptoSwift](https://github.com/krzyzanowskim/CryptoSwift/blob/3cdf5e047030158fafa5747b938e4663db0c07dc/Sources/CryptoSwift/HKDF.swift). I've opened an issue to support further APIs though. In that issue, we should make those structs closer to the APIs that define them.

Reply:
Ah! Alright. Thanks for clearing up my confusion 👍